src/eureka/S3_data_reduction/nircam.py

```diff
@@ -48,16 +50,31 @@ def read(filename, data, meta, log):
     data.attrs['filename'] = filename
     data.attrs['mhdr'] = hdulist[0].header
     data.attrs['shdr'] = hdulist['SCI', 1].header
+
     data.attrs['intstart'] = data.attrs['mhdr']['INTSTART']-1
     data.attrs['intend'] = data.attrs['mhdr']['INTEND']
 
     sci = hdulist['SCI', 1].data
     err = hdulist['ERR', 1].data
     dq = hdulist['DQ', 1].data
     v0 = hdulist['VAR_RNOISE', 1].data
-    wave_2d = hdulist['WAVELENGTH', 1].data
     int_times = hdulist['INT_TIMES', 1].data
 
+    if hdulist[0].header['CHANNEL'] == 'LONG':  # Spectroscopy will have "LONG" as CHANNEL
+        meta.photometry = False
+        wave_2d = hdulist['WAVELENGTH', 1].data
+    elif hdulist[0].header['CHANNEL'] == 'SHORT':  # Photometry will have "SHORT" as CHANNEL
```

Comment:
> Is there no way to get spectropscopy in the short channel using some different observing setup? If there is, then this would break for those data. There must be some other setting in the FITS header which says whether you're looking at spectra or photometry.

That's a great point. As far as I know, NIRCam's Grisms are only in the 2.4–5.0 µm area (so Long Wavelength) using either F277W, F322W2, F356W or F444W. So ['CHANNEL'] == 'SHORT' (i.e. wavelength <= 2.3 micron) will **always** be photometry. But I guess someone could use **just** use the LW light to do photometry and don't do spectroscopy at all? 

So I think the most robust solution would be to check hdulist[0].header['FILTER'] and if that's one of the following: F277W, F322W2, F356W or F444W it **has** to be spectroscopy. In any other case, it's photometry.

Here are some helpful websites:
- https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-instrumentation/nircam-grisms
- https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-instrumentation/nircam-filters